tests/e2e/scenarios/expose-pods-by-name/test.yml

```
@@ -0,0 +1,337 @@
+---
+- name: Skupper Expose-Pods-By-Name Test
+  hosts: all
+  connection: local
+  gather_facts: false
+
+  vars:
+    kubeconfig: kubeconfig
+    search_character: "-"
+
+  tasks:
+    - name: "[ ExposePodsByNameTest ] - Main test block"
+      block:
+        - name: "[ ExposePodsByNameTest - Setup ] Generating the namespaces"
+          ansible.builtin.include_role:
+            name: e2e.tests.generate_namespaces
+
+        - name: "[ ExposePodsByNameTest - Setup ] Creating Skupper resources on east namespace"
+          skupper.v2.resource:
+            kubeconfig: "{{ kubeconfig }}"
+            path: "{{ item }}"
+            namespace: "{{ namespace_prefix }}-{{ namespace_name }}"
+          with_items:
+            - "{{ playbook_dir }}/resources/east/backend.yaml"
+            - "{{ playbook_dir }}/resources/east/site.yaml"
+            - "{{ playbook_dir }}/resources/east/connector.yaml"
+          when:
+            - "'east' in inventory_hostname"
+
+        - name: "[ ExposePodsByNameTest - Setup ] Creating Skupper resources on west namespace"
+          skupper.v2.resource:
+            kubeconfig: "{{ kubeconfig }}"
+            path: "{{ item }}"
+            namespace: "{{ namespace_prefix }}-{{ namespace_name }}"
+          with_items:
+            - "{{ playbook_dir }}/resources/west/site.yaml"
+            - "{{ playbook_dir }}/resources/west/listener.yaml"
+          when:
+            - "'west' in inventory_hostname"
+
+        - name: "[ ExposePodsByNameTest - Setup ] Issue a Skupper access token from west namespace"
+          skupper.v2.token:
+            kubeconfig: "{{ kubeconfig }}"
+            namespace: "{{ namespace_prefix }}-{{ namespace_name }}"
+            name: west-xpbn-grant
+            redemptions_allowed: 2
+          register: west
+          when:
+            - "'west' in inventory_hostname"
+
+        - name: "[ ExposePodsByNameTest - Setup ] Apply token to east site"
+          skupper.v2.resource:
+            kubeconfig: "{{ kubeconfig }}"
+            def: "{{ hostvars['west']['west']['token'] }}"
+            namespace: "{{ namespace_prefix }}-{{ namespace_name }}"
+          when:
+            - "'east' in inventory_hostname"
+
+        - name: "[ ExposePodsByNameTest - Setup ] Wait for links to get in Ready state"
+          kubernetes.core.k8s_info:
+            api_version: skupper.io/v2alpha1
+            kind: Link
+            namespace: "{{ namespace_prefix }}-{{ namespace_name }}"
+            kubeconfig: "{{ kubeconfig }}"
+          register: link_list
+          until: link_list.resources[0].status.status == "Ready"
```

Maybe something like this?
```suggestion
          until: link_list.resources | length > 0 and link_list.resources[0].status.status == "Ready"
```

---

ok, changed